### PR TITLE
Eliminate code moved upstream to Ember Core.

### PR DESCRIPTION
**Pull request by dgeb**
The following has been moved upstream:

* Most registrations
* Boot + initializer concerns
* Engine instance spawning
* Engine dependency cloning

In addition, the following changes have been cherry picked from the v0.2 branch:

* Support engine configuration settings.
* Updates to the README.

### CODE REVIEW

**Comment by dgeb** [comment]
@rwjblue r?

File: README.md [hunk 1]
```diff
@@ -94,14 +95,32 @@ module.exports = EngineAddon.extend({
...
+  return ENV;
+};
+```
+
 Within your engine's `addon` directory, add a new `engine.js` file:
 
 ```js
 import Engine from 'ember-engines/engine';
-import Resolver from 'ember-engines/resolver';
+import Resolver from 'ember-engines/resolver'; // <=== IMPORTANT - custom resolver!!!
```
Comment (rwjblue) [inline README.md]: As a complete aside, we should probably try to help folks deal with this "trollage" a bit and assert if the app isn't using our resolver.

**Comment by rwjblue** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (dgeb) [inline README.md]: Good call. I opened #134 to track.

**Pull request merged**